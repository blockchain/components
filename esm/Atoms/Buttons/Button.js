function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  display: flex;\n  flex-direction: row;\n  justify-content: center;\n  align-items: center;\n  width: 100%;\n  height: 40px;\n  font-size: 18px;\n  color: ", ";\n  background-color: ", ";\n  border: none;\n  border-radius: 5px;\n  cursor: ", ";\n  outline: none;\n  transition-duration: 0.3s;\n  transition-timing-function: ease-in-out;\n  opacity: ", ";\n  font-family: ", ";\n\n  &:hover {\n    background-color: ", ";\n    transform: ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

var getHoverColor = function getHoverColor(disabled, theme) {
  return disabled ? theme['buttonBackgroundColor'] : theme['buttonHoverColor'];
};

var Wrapper = styled.button.attrs({
  type: function type(props) {
    return props.type;
  }
})(_templateObject(), function (props) {
  return props.theme['buttonColor'];
}, function (props) {
  return props.theme['buttonBackgroundColor'];
}, function (props) {
  return props.disabled ? 'not-allowed' : 'pointer';
}, function (props) {
  return props.disabled ? 0.5 : 1;
}, function (props) {
  return props.theme['fontPrimary'];
}, function (props) {
  return getHoverColor(props.disabled, props.theme);
}, function (props) {
  return props.bounced ? 'scale(0.95)' : 'none';
});

var ContainedButton = function ContainedButton(_ref) {
  var children = _ref.children,
      rest = _objectWithoutProperties(_ref, ["children"]);

  return React.createElement(Wrapper, rest, children);
};

ContainedButton.propTypes = {
  bounced: PropTypes.bool,
  children: PropTypes.node,
  disabled: PropTypes.bool,
  type: PropTypes.oneOf(['button', 'reset', 'submit'])
};
ContainedButton.defaultProps = {
  bounced: false,
  disabled: false,
  type: 'button'
};
export default ContainedButton;