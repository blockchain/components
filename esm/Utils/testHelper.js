import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}