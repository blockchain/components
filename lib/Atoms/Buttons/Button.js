"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  display: flex;\n  flex-direction: row;\n  justify-content: center;\n  align-items: center;\n  width: 100%;\n  height: 40px;\n  font-size: 18px;\n  color: ", ";\n  background-color: ", ";\n  border: none;\n  border-radius: 5px;\n  cursor: ", ";\n  outline: none;\n  transition-duration: 0.3s;\n  transition-timing-function: ease-in-out;\n  opacity: ", ";\n  font-family: ", ";\n\n  &:hover {\n    background-color: ", ";\n    transform: ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var getHoverColor = function getHoverColor(disabled, theme) {
  return disabled ? theme['buttonBackgroundColor'] : theme['buttonHoverColor'];
};

var Wrapper = _styledComponents.default.button.attrs({
  type: function type(props) {
    return props.type;
  }
})(_templateObject(), function (props) {
  return props.theme['buttonColor'];
}, function (props) {
  return props.theme['buttonBackgroundColor'];
}, function (props) {
  return props.disabled ? 'not-allowed' : 'pointer';
}, function (props) {
  return props.disabled ? 0.5 : 1;
}, function (props) {
  return props.theme['fontPrimary'];
}, function (props) {
  return getHoverColor(props.disabled, props.theme);
}, function (props) {
  return props.bounced ? 'scale(0.95)' : 'none';
});

var ContainedButton = function ContainedButton(_ref) {
  var children = _ref.children,
      rest = _objectWithoutProperties(_ref, ["children"]);

  return _react.default.createElement(Wrapper, rest, children);
};

ContainedButton.propTypes = {
  bounced: _propTypes.default.bool,
  children: _propTypes.default.node,
  disabled: _propTypes.default.bool,
  type: _propTypes.default.oneOf(['button', 'reset', 'submit'])
};
ContainedButton.defaultProps = {
  bounced: false,
  disabled: false,
  type: 'button'
};
var _default = ContainedButton;
exports.default = _default;