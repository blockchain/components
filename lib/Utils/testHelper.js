"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}