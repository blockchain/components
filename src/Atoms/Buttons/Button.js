// @flow strict
import PropTypes from 'prop-types'
import * as React from 'react'
import styled from 'styled-components'

import injectE2E from '../../Tools/injectE2E'

type PropsType = {
  +bounced?: boolean,
  +children?: React.Node,
  +disabled?: boolean,
  +e2e?: string,
  +type: 'button' | 'reset' | 'submit',
}

const Wrapper = styled.button.attrs({
  'data-e2e': injectE2E,
  type: (props) => props.type,
})`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  height: 2.5rem;
  line-height: 2.5rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  color: ${(props) => props.theme['buttonColor']};
  background-color: ${(props) => props.theme['buttonBackgroundColor']};
  border: none;
  border-radius: ${(props) => props.theme.buttonBorderRadius};
  cursor: ${(props) => (props.disabled ? 'not-allowed' : 'pointer')};
  outline: none;
  transition: all 0.3s ease 0s;
  opacity: ${(props) => (props.disabled ? 0.5 : 1)};
  font-family: ${(props) => props.theme['fontPrimary']};
  font-size: 1rem;
  font-weight: 600;

  &:hover {
    background-color: ${(props) =>
      props.disabled
        ? props.theme['buttonBackgroundColor']
        : props.theme['buttonHoverColor']};
    transform: ${(props) => (props.bounced ? 'scale(0.95)' : 'none')};
  }
`
const Button = ({ children, ...rest }: PropsType) => (
  <Wrapper {...rest}>{children}</Wrapper>
)

Button.propTypes = {
  bounced: PropTypes.bool,
  children: PropTypes.node,
  disabled: PropTypes.bool,
  e2e: PropTypes.string,
  type: PropTypes.oneOf(['button', 'reset', 'submit']),
}

Button.defaultProps = {
  bounced: false,
  disabled: false,
  e2e: 'defaultButton',
  type: 'button',
}

export default Button
