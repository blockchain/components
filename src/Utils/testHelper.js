import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
