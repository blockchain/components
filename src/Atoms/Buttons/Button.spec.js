import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import Button from './Button'

describe('Button', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<Button />)

    expect(tree.find(Button)).toMatchSnapshot()
  })

  it('should have a bounced state', () => {
    const tree = mountWithTheme(<Button bounced />)

    expect(tree.find(Button)).toMatchSnapshot()
  })

  it('should have a disabled state', () => {
    const tree = mountWithTheme(<Button disabled />)

    expect(tree.find(Button)).toMatchSnapshot()
  })

  it('should have a type property', () => {
    const submit = mountWithTheme(<Button type="submit" />)

    expect(submit.find(Button)).toMatchSnapshot()

    const reset = mountWithTheme(<Button type="reset" />)

    expect(reset.find(Button)).toMatchSnapshot()
  })
})
