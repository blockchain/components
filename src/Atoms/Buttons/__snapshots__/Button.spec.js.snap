// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should have a bounced state 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 2.5rem;
  line-height: 2.5rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  color: #FFF;
  background-color: #004A7C;
  border: none;
  border-radius: 0.125rem;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 0.3s ease 0s;
  transition: all 0.3s ease 0s;
  opacity: 1;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  font-weight: 600;
}

.c0:hover {
  background-color: #0068af;
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
}

<Button
  bounced={true}
  disabled={false}
  e2e="defaultButton"
  type="button"
>
  <styled.button
    bounced={true}
    disabled={false}
    e2e="defaultButton"
    type="button"
  >
    <button
      className="c0"
      data-e2e="defaultButton"
      disabled={false}
      type="button"
    />
  </styled.button>
</Button>
`;

exports[`Button should have a disabled state 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 2.5rem;
  line-height: 2.5rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  color: #FFF;
  background-color: #004A7C;
  border: none;
  border-radius: 0.125rem;
  cursor: not-allowed;
  outline: none;
  -webkit-transition: all 0.3s ease 0s;
  transition: all 0.3s ease 0s;
  opacity: 0.5;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  font-weight: 600;
}

.c0:hover {
  background-color: #004A7C;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

<Button
  bounced={false}
  disabled={true}
  e2e="defaultButton"
  type="button"
>
  <styled.button
    bounced={false}
    disabled={true}
    e2e="defaultButton"
    type="button"
  >
    <button
      className="c0"
      data-e2e="defaultButton"
      disabled={true}
      type="button"
    />
  </styled.button>
</Button>
`;

exports[`Button should have a type property 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 2.5rem;
  line-height: 2.5rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  color: #FFF;
  background-color: #004A7C;
  border: none;
  border-radius: 0.125rem;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 0.3s ease 0s;
  transition: all 0.3s ease 0s;
  opacity: 1;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  font-weight: 600;
}

.c0:hover {
  background-color: #0068af;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

<Button
  bounced={false}
  disabled={false}
  e2e="defaultButton"
  type="submit"
>
  <styled.button
    bounced={false}
    disabled={false}
    e2e="defaultButton"
    type="submit"
  >
    <button
      className="c0"
      data-e2e="defaultButton"
      disabled={false}
      type="submit"
    />
  </styled.button>
</Button>
`;

exports[`Button should have a type property 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 2.5rem;
  line-height: 2.5rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  color: #FFF;
  background-color: #004A7C;
  border: none;
  border-radius: 0.125rem;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 0.3s ease 0s;
  transition: all 0.3s ease 0s;
  opacity: 1;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  font-weight: 600;
}

.c0:hover {
  background-color: #0068af;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

<Button
  bounced={false}
  disabled={false}
  e2e="defaultButton"
  type="reset"
>
  <styled.button
    bounced={false}
    disabled={false}
    e2e="defaultButton"
    type="reset"
  >
    <button
      className="c0"
      data-e2e="defaultButton"
      disabled={false}
      type="reset"
    />
  </styled.button>
</Button>
`;

exports[`Button should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 2.5rem;
  line-height: 2.5rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  color: #FFF;
  background-color: #004A7C;
  border: none;
  border-radius: 0.125rem;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 0.3s ease 0s;
  transition: all 0.3s ease 0s;
  opacity: 1;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  font-weight: 600;
}

.c0:hover {
  background-color: #0068af;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

<Button
  bounced={false}
  disabled={false}
  e2e="defaultButton"
  type="button"
>
  <styled.button
    bounced={false}
    disabled={false}
    e2e="defaultButton"
    type="button"
  >
    <button
      className="c0"
      data-e2e="defaultButton"
      disabled={false}
      type="button"
    />
  </styled.button>
</Button>
`;
